samza-core/src/main/java/org/apache/samza/runtime/LocalContainerRunner.java

```diff
@@ -29,17 +31,25 @@
 import org.apache.samza.config.ShellCommandConfig;
 import org.apache.samza.container.SamzaContainer;
 import org.apache.samza.job.model.JobModel;
+import org.apache.samza.metrics.reporter.Metrics;
+import org.apache.samza.metrics.reporter.MetricsHeader;
+import org.apache.samza.metrics.reporter.MetricsSnapshot;
+import org.apache.samza.serializers.model.SamzaObjectMapper;
+import org.apache.samza.util.FileUtil;
 import org.apache.samza.util.SamzaUncaughtExceptionHandler;
+import org.apache.samza.util.Util;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.slf4j.MDC;
-import java.util.Random;
+
 
 /**
  * Launches and manages the lifecycle for {@link SamzaContainer}s in YARN.
  */
 public class LocalContainerRunner {
   private static final Logger log = LoggerFactory.getLogger(LocalContainerRunner.class);
+  private static final String CONTAINER_METADATA_FILENAME_FORMAT = "%s.metadata"; // Filename: containerID.metadata
+  private static final String CONTAINER_METADATA_DIRECTORY = System.getProperty("samza.log.dir");
```

Comment:
Would prefer moving this to JobConfig instead of referencing system properties directly (they're not as well documented). Should also handle the case where this is missing (e.g. don't do anything).